recipes/zlib-ng/all/conanfile.py

```diff
@@ -0,0 +1,98 @@
+    homepage = "https://github.com/Dead2/zlib-ng/"
+    license ="Zlib"
+    topics = ("conan", "zlib", "compression")
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "zlib_compat": [True, False],
+               "with_gzfileop": [True, False],
+               "with_optim": [True, False],
+               "with_new_strategies": [True, False],
+               "with_native_instructions": [True, False],
+               "fPIC": [True, False]}
+    default_options = {"shared": False,
+                       "zlib_compat": False,
+                       "with_gzfileop": True,
+                       "with_optim": False,
+                       "with_new_strategies": True,
+                       "with_native_instructions": False,
+                       "fPIC": True}
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("zlib-ng-{}".format(self.version), self._source_subfolder)
+
+    def validate(self):
+        if self.options.zlib_compat and not self.options.with_gzfileop:
+            raise ConanInvalidConfiguration("The option 'with_gzfileop' must be True when 'zlib_compat' is True.")
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+        self._cmake.definitions["ZLIB_ENABLE_TESTS"] = False
+        self._cmake.definitions["ZLIB_COMPAT"] = self.options.zlib_compat
+        self._cmake.definitions["WITH_GZFILEOP"] = self.options.with_gzfileop
+        self._cmake.definitions["WITH_OPTIM"] = self.options.with_optim
+        self._cmake.definitions["WITH_NEW_STRATEGIES"] = self.options.with_new_strategies
+        self._cmake.definitions["WITH_NATIVE_INSTRUCTIONS"] = self.options.with_native_instructions
+
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE.md", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "ZLIB"
+        self.cpp_info.names["cmake_find_package_multi"] = "ZLIB"
```

Comment:
https://github.com/zlib-ng/zlib-ng/blob/29fd4672a2279a0368be936d7cd44d013d009fae/CMakeLists.txt#L999

```suggestion
        #FIXME: CMake targets are https://github.com/zlib-ng/zlib-ng/blob/29fd4672a2279a0368be936d7cd44d013d009fae/CMakeLists.txt#L914
        self.cpp_info.names["pkg_config"] = "zlib" + suffix
```

Reply:
how would you fix that? by setting target names conditionally? 
```
    add_library(${library_name} SHARED ${ZLIB_ALL_SRCS})
    add_library(${static_library_name} STATIC ${ZLIB_ALL_SRCS})
```

Reply:
https://github.com/conan-io/conan-center-index/pull/5264#discussion_r618829860

I posted it earlier, you can see the references commit